classes/Category.php

```diff
@@ -1645,16 +1644,6 @@ public function cleanGroups()
         return Db::getInstance()->delete('category_group', 'id_category = ' . (int) $this->id);
     }
 
-    /**
-     * Remove associated products.
-     *
-     * @return bool Indicates whether the cleanup was successful
-     */
-    public function cleanAssoProducts()
```

Comment:
I'm wondering about category deletions that doesn't use the command handlers. I'm not very familiar with the webservice but does it use commands ? @PrestaShop/prestashop-maintainers 
Just calling category->delete() will lead to DB inconsistency. How can we be sure that won't happen ?

Reply:
Very nice catch.